docs/flow/modin/experimental/core/execution/native/implementations/hdk_on_native/index.rst

```diff
@@ -174,30 +166,30 @@ Execution of a computation tree
 
 Frames are materialized (executed) when their data is accessed. E.g. it
 happens when we try to access the frame's index or shape. There are two ways
-to execute required operations: through Arrow or through OmniSciDB.
+to execute required operations: through Arrow or through HDK.
 
 Arrow execution
 '''''''''''''''
 
 For simple operations which don't include actual computations, execution can use
 Arrow API. We can use it to rename columns, drop columns and concatenate
 frames. Arrow execution is preferable since it doesn't require actual data import/export
-to the OmniSciDB.
+from/to HDK.
 
-OmniSciDB execution
-'''''''''''''''''''
+HDK execution
+'''''''''''''
 
-To execute query in OmniSciDB engine we need to import data first. We should
+To execute a query in the HDK engine we need to import data first. We should
 find all leaves of an operation tree and import their Arrow tables. Partitions
 with imported tables hold corresponding table names used to refer to them in
 queries.
 
-OmniSciDB is SQL-based. SQL query parsing is done in a separate process using
-the Apache Calcite framework. A parsed query is serialized into JSON format
-and is transferred back to OmniSciDB. In Modin, we don't generate SQL queries
-for OmniSciDB but use this JSON format instead. Such queries can be directly
-executed by OmniSciDB and also they can be transferred to Calcite server for
-optimizations.
+HDK executes queries in the HDK-specific intermediate representation (IR) format. In Modin we don't generate
```

Comment:
I'd change the wording here to first introduce all formats and their relationship and then write about Modin. Proposed text:
```
HDK executes queries expressed in HDK IR. It also provides components to translate SQL queries to
relational algebra JSON format which can be later optimized and translated to HDK IR. Modin generates
queries in relational algebra JSON format. These queries are optionally optimized with Apache Calcite
based optimizer provided by HDK (:py:class:`~pyhdk.sql.Calcite`) and then executed. 
```